android/src/com/mapswithme/maps/intent/Factory.java

```diff
@@ -250,203 +110,88 @@ public MapTask process(@NonNull final Intent intent)
-    public final boolean isSupported(@NonNull Intent intent)
-    {
-      final Uri data = intent.getData();
-
-      if (data == null)
-        return false;
-
-      String scheme = intent.getScheme();
-      String host = data.getHost();
-
-      return SCHEME_HTTPS.equals(scheme) && (HOST.equals(host) || HOST_DEV.equals(host)) &&
-          isLinkSupported(data);
-    }
-
-    abstract boolean isLinkSupported(@NonNull Uri data);
-
-    @NonNull
-    @Override
-    final MapTask createMapTask(@NonNull String url)
-    {
-      if (isFirstLaunch())
-      {
-        /*
-        MapTask introductionTask = createIntroductionTask(url);
-        if (introductionTask != null)
-          return introductionTask;
-        */
-      }
-
-      return createTargetTask(url);
-    }
-
     @Nullable
-    abstract MapTask createIntroductionTask(@NonNull String url);
-    @NonNull
-    abstract MapTask createTargetTask(@NonNull String url);
-  }
-
-  private static class OpenCountryTaskProcessor implements IntentProcessor
-  {
-    @Override
-    public boolean isSupported(@NonNull Intent intent)
-    {
-      return intent.hasExtra(DownloadResourcesLegacyActivity.EXTRA_COUNTRY);
-    }
-
-    @NonNull
     @Override
     public MapTask process(@NonNull Intent intent)
     {
+      if (!intent.hasExtra(DownloadResourcesLegacyActivity.EXTRA_COUNTRY))
+        return null;
       String countryId = intent.getStringExtra(DownloadResourcesLegacyActivity.EXTRA_COUNTRY);
       return new ShowCountryTask(countryId);
     }
   }
 
-  private static class KmzKmlProcessor implements IntentProcessor
+  public static class KmzKmlProcessor implements IntentProcessor
   {
-    private static final Logger LOGGER = LoggerFactory.INSTANCE.getLogger(LoggerFactory.Type.MISC);
-    private static final String TAG = KmzKmlProcessor.class.getSimpleName();
-    private Uri mData;
     @NonNull
     private final DownloadResourcesLegacyActivity mActivity;
 
-    KmzKmlProcessor(@NonNull DownloadResourcesLegacyActivity activity)
+    public KmzKmlProcessor(@NonNull DownloadResourcesLegacyActivity activity)
     {
       mActivity = activity;
     }
 
-    @Override
-    public boolean isSupported(@NonNull Intent intent)
-    {
-      // See KML/KMZ/KMB intent filters in manifest.
-      if (intent.getAction() == Intent.ACTION_VIEW)
-        mData = intent.getData();
-      else if (intent.getAction() == Intent.ACTION_SEND)
-        mData = intent.getParcelableExtra(Intent.EXTRA_STREAM);
-      return mData != null;
-    }
-
     @Nullable
     @Override
     public MapTask process(@NonNull Intent intent)
     {
+      // See KML/KMZ/KMB intent filters in manifest.
+      Uri data = null;
+      if (intent.getAction().equals(Intent.ACTION_VIEW))
+        data = intent.getData();
+      else if (intent.getAction().equals(Intent.ACTION_SEND))
+        data = intent.getParcelableExtra(Intent.EXTRA_STREAM);
+      if (data == null)
+        return null;
+      final Uri uri = data;
```

Comment:
nit: Leave one variable uri.

Reply:
Fixed. The problem was that a lambda below required "final" variable.